bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/api/DeviceController.java

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.sunsynk.internal.api;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Properties;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.binding.sunsynk.internal.api.dto.APIdata;
+import org.openhab.binding.sunsynk.internal.api.dto.Battery;
+import org.openhab.binding.sunsynk.internal.api.dto.Daytemps;
+import org.openhab.binding.sunsynk.internal.api.dto.Grid;
+import org.openhab.binding.sunsynk.internal.api.dto.RealTimeInData;
+import org.openhab.binding.sunsynk.internal.api.dto.Settings;
+import org.openhab.binding.sunsynk.internal.api.exception.SunSynkDeviceControllerException;
+import org.openhab.binding.sunsynk.internal.api.exception.SunSynkGetStatusException;
+import org.openhab.binding.sunsynk.internal.api.exception.SunSynkSendCommandException;
+import org.openhab.binding.sunsynk.internal.config.SunSynkInverterConfig;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link DeviceController} class defines methods that control
+ * communication with the inverter.
+ *
+ * @author Lee Charlton - Initial contribution
+ */
+
+@NonNullByDefault
+public class DeviceController {
+
+    private final Logger logger = LoggerFactory.getLogger(DeviceController.class);
+    private String sn = "";
+    private String alias = "";
+    private Settings batterySettings = new Settings();
+    private Battery realTimeBattery = new Battery();
+    private Grid grid = new Grid();
+    private Daytemps inverter_day_temperatures = new Daytemps();
```

Comment:
SAT Error: Name 'inverter_day_temperatures' must match pattern '^[a-z][a-zA-Z0-9]*$'.
```suggestion
    private Daytemps inverterDayTemperatures = new Daytemps();
```